### PR TITLE
Handle non-string types in nap.utils.to_unicode

### PR DESCRIPTION
**Pull request by mpistrang**
update `nap.utils.to_unicode` to handle non-string types just like `ctar_lib.cstar.py23.to_unicode` does for 7_00.

### CODE REVIEW

**Comment by mpistrang** [comment]
@neruson @ethurgood @mcordes @cjcooper @afitzgerald Could you please take a look at this PR.  This is causing raven to blow up when attempting to sent errors to sentry since it cannot call repr() on Nap resources.

Disclaimer - not sure if I'm following the correct git flow here, and if somebody could direct me on the right way to get this into 7_00 I would appreciate it.

**Comment by mpistrang** [comment]
Ok I am a bit confused - maybe this is already in master, but the tag 7_00 is not in the right place?   I could def use some help here!

File: nap/utils.py [hunk 1]
```diff
@@ -54,7 +54,20 @@ def flatten_params(k, vs):
 
 
 __text_fn = str if six.PY3 else unicode
```
Comment (neruson) [inline nap/utils.py]: All this is doing is redefining `six.text_type`. Can we remove it?

```
# Python 2
>>> six.text_type is unicode
True

# Python 3
>>> six.text_type is str
True
```

**Comment by neruson** [comment]
I'm getting déjà vu (heyyy unicode) from this PR. Didn't Matt already merge this same change in PR #10?

**Comment by mpistrang** [comment]
I thought he did too, it is in master, but its not in the source that is bootstrapped in with @700:

The following code is on the semiprod proc box in bach_service's virtualenv:

__text_fn = str if six.PY3 else unicode
def to_unicode(s):
    if s is None:
       return None
    return s if isinstance(s, six.text_type) else __text_fn(s, 'utf-8')


Now do we fix the 7_00 tag!?

**Comment by afitzgerald** [comment]
Moving that tag sounds right/familiar. 






On Fri, Jan 15, 2016 at 6:23 AM -0800, "Michael P" <notifications@github.com> wrote:












I thought he did too, it is in master, but its not in the source that is bootstrapped in with @700:



The following code is on the semiprod proc box in bach_service's virtualenv:



__text_fn = str if six.PY3 else unicode

def to_unicode(s):

    if s is None:

       return None

    return s if isinstance(s, six.text_type) else __text_fn(s, 'utf-8')



Now do we fix the 7_00 tag!?

File: nap/utils.py (hunk 1 above)
Comment (cjcooper) [inline nap/utils.py]: Im good with that.

**Comment by mpistrang** [comment]
Ok, I am going to reject this PR.  Alan, can you figure out who can fix what is actually needed here?

**Comment by cjcooper** [comment]
Correct, proper proceedure it not ONLY merge it into master but move the tag associated with the release you are targeting.  If you move 7.00 please move 7.01o and 7.01.

**Pull request closed without merging**